### ISSUE TITLE
Mesh Issues with Large Selections

### ISSUE THREAD

**Issue by zabuss** [jmc2obj/j-mc-2-obj#180]
When exporting very large OBJs (250mb+) from jmc2obj, some areas of mesh don't display properly. For example, in a large export, blocks are 2x as wide as they should be on one axis. This is with the "Optimize Mesh" option turned on. Additionally, some faces are missing from the sides of blocks. I have never had this issue with smaller objs, so I think it is something to do with the size of the obj and the Optimize Mesh option.

**Comment by zabuss** [comment]
here's an example of the wide blocks with some faces missing
<img width="500" alt="Screen Shot 2021-12-11 at 7 37 22 PM" src="https://user-images.githubusercontent.com/90361221/145697255-5fd121ac-5b6d-4504-8c6c-4bd43e3521c4.png">

**Comment by mmdanggg2** [comment]
Does it happen the same with optimise turned off? That kind of looks like it might be floating point prescision errors. Try with exporting it with the model centered too.

**Comment by zabuss** [comment]
Yeah, I exported it with it centered, but it's very hard to export without optimize mesh turned on because of how big the models get

**Comment by zabuss** [comment]
After a bit of testing, the error persists with optimize turned on and off, and Allow Duplicate Vertexes and "Add double sided faces for backface culling"
Additionally, I think it is not a size issue but a dimensions issue, big export selections still have the issue even if there is not many blocks in the selection.

**Comment by mmdanggg2** [comment]
Ok, thanks for checking that. What coordinates is the area you are exporting?

**Comment by zabuss** [comment]
This may be the issue. The coordinates are over 4 million blocks from the origin. (-8,787,612, -4,120,796),(-8,786,254 -4,119,655)

**Comment by mmdanggg2** [comment]
That's the problem then. At 8 milion blocks, floating point precision just can't specify the models properly. I think for now your best bet is to try and move the region you want to export closer to 0 0 coordinates with another tool. I will have a look at increasing to double precision since this isn't something that is too out of the ordinary for a minecraft world.

**Comment by zabuss** [comment]
Ok, good to know. I’ll close the issue.
